src/client/js/client.js

```diff
@@ -151,6 +152,13 @@ define([
                         logger.warn('Lots of persisted objects', numberOfPersistedObjects);
                     }
 
+                    if (gmeConfig.storage.patchRootCommunicationEnabled) {
+                        persisted.rootHash = {

```

Comment:
Couldn't these be provided from the core?
So when enabled, instead of returning just `{rootHash: '#..', objects: {}}`, it also adds rootInfo?